pilot/pkg/kube/inject/inject.go

```diff
@@ -540,6 +540,24 @@ func flippedContains(needle, haystack string) bool {
 func InjectionData(sidecarTemplate, valuesConfig, version string, deploymentMetadata *metav1.ObjectMeta, spec *corev1.PodSpec,
 	metadata *metav1.ObjectMeta, proxyConfig *meshconfig.ProxyConfig, meshConfig *meshconfig.MeshConfig) (
 	*SidecarInjectionSpec, string, error) {
+
+	var name string
+	if metadata.Namespace == "" {
+		if deploymentMetadata.Namespace != "" {
+			name = deploymentMetadata.Namespace + "/" + metadata.Name
+		} else {
+			name = metadata.Name
+		}
+	} else {
+		name = metadata.Namespace + "/" + metadata.Name
+	}
+
+	// If DNSPolicy is not ClusterFirst, the Envoy sidecar may not able to connect to Istio Pilot.
+	if spec.DNSPolicy != corev1.DNSClusterFirst {
+		log.Warnf("%q's DNSPolicy is not %q. The Envoy sidecar may not able to connect to Istio Pilot\n",
```

Comment:
redundant `\n`, s/name/metadata.Namespace + "/" + metadata.Name

Reply:
The namespace may be "", which makes the result be `"/test"'s DNSPolicy ...`.

Reply:
I know that, it only occurs using istioctl, doesn't matter much